assets/app/scripts/services/data.js

```diff
@@ -159,6 +159,92 @@ angular.module('openshiftConsole')
     }
   	return deferred.promise;
   };
+  
+// type:      API type (e.g. "pods")
+// object:    API object data(eg. { kind: "Build", parameters: { ... } } )
+// context:   API context (e.g. {project: "..."})
+// opts:      http - options to pass to the inner $http call
+// Returns a promise resolved with response data or rejected with {data:..., status:..., headers:..., config:...} when the delete call completes.
+  DataService.prototype.create = function(type, object, context, opts) {
+  	opts = opts || {};
+  	var deferred = $q.defer();
+    if (context.projectPromise && type !== "projects") {
+      var self = this;
+      context.projectPromise.done(function(project) {
+        $http(angular.extend({
+          method: 'POST',
+          data: object,
+          url: self._urlForType(type, null, context, false, {namespace: project.metadata.name})
+        }, opts.http || {}))
+        .success(function(data, status, headerFunc, config, statusText) {
+          deferred.resolve(data);
+        })
+        .error(function(data, status, headers, config) {
+          deferred.reject({
+            data: data,
+            status: status,
+            headers: headers,
+            config: config
+          });
+        });
+      });
+    }
+    else {
+      $http(angular.extend({
+        method: 'POST',
+        data: object,
+        url: this._urlForType(type, null, context)
+      }, opts.http || {}))
+      .success(function(data, status, headerFunc, config, statusText) {
+        deferred.resolve(data);
+      })
+      .error(function(data, status, headers, config) {
+        deferred.reject({
+          data: data, 
+          status: status, 
+          headers: headers, 
+          config: config
+        });
+      });
+    }
+  	return deferred.promise;
+  };
+  
+
+  // objects:   Array of API object data(eg. [{ kind: "Build", parameters: { ... } }] )
+  // context:   API context (e.g. {project: "..."})
+  // opts:      http - options to pass to the inner $http call
+  // Returns a promise resolved with an an object like: { success: {}, failure: {} }
+  // where success and failure contain an array of 
+  DataService.prototype.createList = function(objects, context, opts) {
+    var result = $q.defer();
+    var success_results = []
```

Comment:
lots of dropped semi-colons on end of lines in this method.  We are trying to keep ; for consistency